site/quorum-queue.md

```diff
@@ -0,0 +1,267 @@
+quorum queues will have to be manually adjusted using the `rabbitmq-queues`
+command line tool.
+
+Two commands are provided:
+
+`rabbitmq-queues add_member [-p <vhost>] <queue-name> <node>`
+
+`rabbitmq-queues delete_member [-p <vhost>] <queue-name> <node>`
+
+NB: to add and remove members a quorum needs to be available.
+
+Care needs to be taken not to accidentally make a queue unavailable by losing
+the quorum whilst performing membership changes.
+
+
+## Behaviour to expect
+
+Quorum queues are quorum systems and use consensus to ensure data consistency.
+
+#### [Fault tolerance](#fault-tolerance)
+
+Consensus systems can provide certain guarantees with regard to data safety.
+These guarantees do mean that certain conditions need to be met before they
+become relevant such as requiring a minimum of 3 RabbitMQ nodes to provide
+fault tolerance and requiring more than half of members to be available to
+work at all.
+
+*Failure tolerance table*:
+
+| Node count | Tolerated no. of failures |
+| :--------: | :------: |
+| 1          | 0        |
+| 2          | 0        |
+| 3          | 1        |
+| 4          | 1        |
+| 5          | 2        |
+| 6          | 2        |
+| 7          | 3        |
+
+As the table above shows RabbitMQ clusters with fewer than 3 nodes do not
+benefit fully from the quorum queue guarantees. RabbitMQ clusters with an even
+number of RabbitMQ nodes do not benefit from having quorum queue members spread
+over all nodes. For these systems the quorum queue size should be constrained to a
+smaller uneven number of nodes.
+
+Performance tails off quite a bit for quorum queue node sizes larger than 5.
+We do not recommend running quorum queues on more than 7 RabbitMQ nodes. The
+default quorum queue size is 5 and is controllable using the
+`x-quorum-queue-size` queue argument.
+
+#### [Data safety](#data-safety)
+
+Quorum queues are designed to provide data safety under network partition and
+failure scenarios. A message that was successfully confirmed back to the publisher
+using the [publisher confirms](confirms.html) feature should not be lost as long as at
+least a majority of RabbitMQ nodes hosting the quorum queue are not
+permanently made unavailable.
+
+Generally quorum queues favours data consistency over availability.
+
+*_No guarantees are provided for messages that have not been confirmed using
+the publisher confirm mechanism_*. They may never reach the queue.
+
+
+#### [Availability](#availability)
+
+Quorum queues should be able to tolerate a minority of queue members becoming unavailable
+with no or little affect on availability. (NB: RabbitMQ may restart itself during
+recovery and perform various actions that make this harder to ensure but in
+principle this should be true).
+
+If a quorum of nodes cannot be recovered (say if 2 out of 3 RabbitMQ nodes are
+permanently lost) the queue is permanently unavailable and
+will need to be force deleted and recreated.
+
```

Comment:
This new queue type will be used to store data, hence the queue depth can be high.
There should be a section on resource (disk and memory) usage and a warning that you can't have infinite queue because there is no flushing to disk.